zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -295,8 +295,8 @@ static boolean isValid(int type) {
         // make sure this is always synchronized with Zoodefs!!
         switch (type) {
         case OpCode.notification:
-            return false;
         case OpCode.check:
+            return false;
```

Comment:
This is the effective change beyond the revert. According to https://lists.apache.org/thread/vl816jfrklvqz29coz5qnwpom9q41pcg.